packages/fe-fpm-writer/src/action/types.ts

```diff
@@ -25,8 +46,9 @@ export interface CustomAction extends CustomElement {
         /**
          * If not set (i.e. undefined) then no event handler is linked. If it is set true, a new one is created and linked to the action.
          * If an existing event handler is to be used then its id needs to be provided as string.
+         * Object with 'CustomActionNewEventHandler' provides option to append existing js file with new action.
          */
-        eventHandler?: string | true;
```

Comment:
Should we just change the type or allow `string | true | CustomActionEventHandler` and be backward compatible? I am ok with replacing it, since we only have one known consumer, so changing it now is easy. Just wanted to have it mentioned and discussed.

Reply:
Hi @tobiasqueck 

I rechecked type and I think `string | CustomActionEventHandler` fits well.
By providing string we can simply set full path to existing method.
And if we need to create new js file or append existing js file with additional code fragment, then we can use object with `CustomActionNewEventHandler` interface.

Interface could look following:
```
export interface CustomActionNewEventHandler {
    /**
     * JS file name - existing or new.
     */
    fileName?: string;
    /**
     * Function name. If undefined then `onPress` is used as default value.
     */
    fnName?: string;
    /**
     * If file exists, then existing file should be appended with passed script fragment.
     */
    insertScript?: {
        /**
         * Insert position for passed script fragment.
         */
        position: FileContentPosition;
        /**
         * Full fragment of script to insert.
         */
        content: string;
    };
}

export type CustomActionEventHandler = string | true | CustomActionNewEventHandler;
```

Some examples would be:
1. New event handler(js file and method) with default file and function name:
```
{
    name: 'dummy',
    eventHandler: {},
    folder: join('ext', 'fragments')
}
```
New file at `ext/fragments/Dummy.js` with `onPress` will be created.

2. New event handler with custom function and file name:
```
{
    name: 'dummy',
    eventHandler: {
        fnName: 'onTest',
        fileName: 'newAction.js',
    },
    folder: join('ext', 'fragments')
}
```
New file at `ext/fragments/newAction.js` will be created. And in manifest we will have:
```
Dummy": {
  "press": "project4.ext.fragment.newAction.onPress",
  ...
}
```

3. Append script fragment to existing file(ext/fragments/MyExistingAction.js):
```
{
    eventHandler: {
        fnName: 'onHandleSecondAction',
        fileName: 'MyExistingAction.js',
        insertScript: {
            content: `,\n    onHandleSecondAction: function(){\n      console.log();\n    }`,
            position: {
                line: 8,
                character: 9
            }
        }
    },
    folder: join('ext', 'fragments')
}
```
`MyExistingAction.js` would be appended with additional js code fragment.

4. Use existing event handler:
```
{
    name: 'dummy',
    eventHandler: 'project4.ext.fragment.newAction.onPress',
    folder: join('ext', 'fragments')
}
```
No new or existing js files modified and entry in manifest would be:
```
Dummy": {
  "press": "project4.ext.fragment.newAction.onPress",
  ...
}
```

Is such interface ok for you?

Reply:
@tobiasqueck 

I rechecked interface again and I think `true | string | EventHandlerConfiguration` fits well.
I think we need `string` for sure - in case to provide existing without any change in target js files.

We can get rid of `true` and use `eventHandler: {}` to setup handler with default values like:
1. file name by default equals to `config.name`;
2. function name in handler by default equals to `onPress`;

We can provide custom values by using properties `fileName` or `fnName` like:
```
eventHandler: {
    fnName: 'onPress',
    fileName: config.name
}
```
But I think it is better to have `fileName` and `fnName` as optional properties, which leads that shortest way to create handler is `eventHandler: {}`, which would be equal to `eventHandler: true`. So I think in result there no much benefit to remove `true` from `eventHandler?: true | string | EventHandlerConfiguration`.

Please correct if you have different interface in mind.